splice_machine/src/main/java/com/splicemachine/derby/utils/Vacuum.java

```diff
@@ -132,72 +145,35 @@ public void vacuumDatabase() throws SQLException{
         }
     }
 
-    /*
-     * We have to make sure that all prior transactions complete. Once that happens, we know that the worldview
-     * of all outstanding transactions is the same as ours--so if a conglomerate doesn't exist in sysconglomerates,
-     * then it's not useful anymore.
-     */
-    private void ensurePriorTransactionsComplete() throws SQLException {
+    private long getOldestActiveTransaction() throws StandardException {
         EmbedConnection embedConnection = (EmbedConnection)connection;
 
         TransactionController transactionExecute = embedConnection.getLanguageConnection().getTransactionExecute();
-        TxnView activeStateTxn = ((SpliceTransactionManager) transactionExecute).getActiveStateTxn();
-
-        //wait for all transactions prior to us to complete, but only wait for so long
-        try{
-            long activeTxn = waitForConcurrentTransactions(activeStateTxn);
-            if(activeTxn>0){
-                //we can't do anything, blow up
-                throw PublicAPI.wrapStandardException(
-                        ErrorState.DDL_ACTIVE_TRANSACTIONS.newException("VACUUM", activeTxn));
-            }
-
-        }catch(StandardException se){
-            throw PublicAPI.wrapStandardException(se);
-        }
-    }
+        TxnView txn = ((SpliceTransactionManager) transactionExecute).getActiveStateTxn();
 
-    private long waitForConcurrentTransactions(TxnView txn) throws StandardException {
         ActiveTransactionReader reader = new ActiveTransactionReader(0l,txn.getTxnId(),null);
-        SConfiguration config = EngineDriver.driver().getConfiguration();
-        long timeRemaining = config.getDdlDrainingMaximumWait();
-        long pollPeriod = config.getDdlDrainingInitialWait();
-        int tryNum = 1;
+
         long activeTxn;
 
         try {
-            do {
-                activeTxn = -1l;
-
-                TxnView next;
-                try (Stream<TxnView> activeTransactions = reader.getActiveTransactions()){
-                    while((next = activeTransactions.next())!=null){
-                        long txnId = next.getTxnId();
-                        if(txnId!=txn.getTxnId()){
-                            activeTxn = txnId;
-                            break;
-                        }
+            activeTxn = txn.getEffectiveBeginTimestamp();
+
+            TxnView next;
+            try (Stream<TxnView> activeTransactions = reader.getActiveTransactions()){
+                while((next = activeTransactions.next())!=null){
+                    long txnId = next.getEffectiveBeginTimestamp();
+                    if(txnId < activeTxn){
+                        activeTxn = txnId;
+                        break;
```

Comment:
The 'if' clause breaks out of the while loop without searching all of the active transactions, so it might not find the oldest active transaction.